src/lib/adapters/waku/safe-waku.ts

```diff
@@ -275,10 +286,27 @@ export class SafeWaku {
 		this.subscriptions.set(chatId, subscription)
 	}
 
+	private areMessagesEqual(a: Message, b: Message): boolean {
+		if (a.timestamp !== b.timestamp) {
+			return false
+		}
+		if (a.type !== b.type) {
+			return false
+		}
+		if (a.type === 'babble' && b.type === 'babble' && a.id !== b.id) {
+			return false
+		}
+		if (a.type !== 'babble' && b.type !== 'babble' && a.senderPublicKey !== b.senderPublicKey) {
+			return false
+		}
+
+		return true
+	}
+
```

Comment:
This is to avoid including mesages twice? Why don't we check the content as well?

Reply:
Good catch! Now that we have metadata it is enough to check the equality for `timestamp`, `id` and `senderPublicKey`. I added a fix for this.